public/app/features/dashboard/dashgrid/DashboardGrid.tsx

```diff
@@ -153,8 +164,11 @@ export class DashboardGrid extends PureComponent<Props> {
 
     this.props.dashboard.sortPanelsByGridPos();
 
-    // Call render() after any changes.  This is called when the layout loads
-    this.forceUpdate();
+    // Call render() after any changes.  This is called when the initial layout loads
+    if (!this.state.isLayoutInitialized) {
+      this.forceUpdate();
+      this.setState({ isLayoutInitialized: true });
+    }
```

Comment:
Can you describe why new condition exists? dont we always need to re-render onLayoutChange ?

Reply:
Not really, `onLayoutChange` is meant to be a mechanism to save the layout after every drag or resize stop. The library itself should take care of the re-rendering part if we update the `layout` object, I think this condition was here before because of this bug https://github.com/react-grid-layout/react-grid-layout/issues/670  (it was solved in early versions), check this comment https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/dashgrid/DashboardGrid.tsx#L167 , yet we still need to keep forceUpdate when the layout is initially built (I suspect we are doing something with lazy loading or something because when I removed this `forceUpdate` it was rendering an empty dashboard)

Reply:
Oh, I forgot to mention, removing this condition will re-render the panel twice when we update the `panel.gridPos` via **InspectJSONTab**, the first one because the event triggers a `forceUpdate`, and the second one because of this line on the `onLayoutChange` method